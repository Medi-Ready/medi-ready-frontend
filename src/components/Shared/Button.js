import React from "react";

import PropTypes from "prop-types";
import styled from "styled-components";

const Button = ({ type, className, children, onClick }) => {
  return (
    <DefaultButton type={type} className={className} onClick={onClick}>
      {children}
    </DefaultButton>
  );
};

Button.propTypes = {
  type: PropTypes.string.isRequired,
  children: PropTypes.string.isRequired,
};

Button.defaultProps = {
  type: "submit",
};

const DefaultButton = styled.button`
  display: inline-block;
  min-width: 80px;
  height: 45px;
  border-radius: 8px;
  background-color: #2f49d1;
  vertical-align: middle;
  color: #fff;
`;

export default Button;
