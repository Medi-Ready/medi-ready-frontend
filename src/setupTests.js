import React from "react";
import ReactDOM from "react-dom";
import "@testing-library/jest-dom";

import { Router } from "react-router-dom";
import { createMemoryHistory } from "history";
import { render } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import { QueryClient, QueryClientProvider } from "react-query";

import theme from "../src/styles/theme";
import GlobalStyles from "../src/styles/";
import { ModalProvider } from "../src/contexts/ModalContext";

const queryClient = new QueryClient();

export const history = createMemoryHistory({
  initialEntries: ["/"],
});

export const customRender = (children) => (
  render((
    <Router history={history}>
      <ThemeProvider theme={theme}>
        <QueryClientProvider client={queryClient}>
          <GlobalStyles />
          <ModalProvider>
            {children}
          </ModalProvider>
        </QueryClientProvider>
      </ThemeProvider>
    </Router>
  ))
);

beforeAll(() => {
  ReactDOM.createPortal = jest.fn((element) => element);
});
