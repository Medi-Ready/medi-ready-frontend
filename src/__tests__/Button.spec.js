import React from "react";
import { screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import { customRender } from "../setupTests";
import Button from "../components/Shared/Button";

describe("<Button />", () => {
  it("matches snapshot", () => {
    const { container } = customRender(
      <Button type="button" onClick={() => console.log("test")}>test</Button>
    );

    expect(container).toMatchSnapshot();
  });

  it("should render without crashing", () => {
    const { getByText } = customRender(
      <Button type="button" onClick={() => console.log("test")}>test</Button>,
    );

    expect(getByText("test")).toBeInTheDocument();
  });

  it("should occur an event", () => {
    jest.spyOn(global.console, "log");

    customRender(
      <Button type="button" onClick={() => console.log("click")}>test</Button>,
    );

    const button = screen.getByText("test");

    userEvent.click(button);

    expect(console.log).toHaveBeenCalledWith("click");
  });
});
