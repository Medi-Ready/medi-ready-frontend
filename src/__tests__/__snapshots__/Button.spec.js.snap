// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> matches snapshot 1`] = `
<div>
  <button
    class="sc-hKgJUU kstYDR"
    type="button"
  >
    test
  </button>
</div>
`;
